### PR TITLE
UIREC-28 Add Title for receiving/check-in

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
## Purpose
Form for add Title
https://issues.folio.org/browse/UIREC-28

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/resources/titleResources.js [hunk 1]
```diff
@@ -9,3 +9,12 @@ export const titlesResource = {
...
 };
+
+export const identifierTypesResource = {
+  ...baseManifest,
+  path: 'identifier-types',
+  params: {
+    query: 'cql.allRecords=1 sortby name',
+  },
+  records: 'identifierTypes',
+};
```
Comment (AndreiShumski2) [inline src/common/resources/titleResources.js]: Do you need to add `okapiInterface` version to `package.json` for `identifierTypes`?

File: src/TitleForm/TitleForm.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                        disabled
+                        label={<FormattedMessage id="ui-receiving.titles.title" />}
+                        name="title"
+                        required
+                        type="text"
+                        validate={validateRequired}
+                      />
+                      <OnChange name="title">
+                        {(value, previous) => {
+                          console.log('OnChange title', value, previous);
```
Comment (NikitaSedyx) [inline src/TitleForm/TitleForm.js]: is it required?

File: src/TitleForm/TitleForm.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  useAccordionToggle,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  SECTIONS,
+} from './constants';
+import ProductIdDetailsForm from './ProductIdDetailsForm';
+
+const FormFooter = ({
```
Comment (NikitaSedyx) [inline src/TitleForm/TitleForm.js]: I remember someone did common component, but I couldn't find in acq-components

File: src/TitleForm/TitleFormContainer.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      return mutator.titles.POST(newTitle)
+        .then(savedTitle => {
+          setTimeout(() => history.push('/receiving'));
+        });
+    },
+    [history, titleId],
+  );
+
+  if (!identifierTypes) {
+    return null;
```
Comment (AndreiShumski2) [inline src/TitleForm/TitleFormContainer.js]: Why not `<LoadingPane>`?

File: src/Receiving.js [hunk 5]
```diff
@@ -3,15 +3,28 @@ import {
...
+      <Route
+        component={ReceivingListContainer}
+        path="/receiving"
+      />
     </Switch>
   );
 };
 
+Receiving.propTypes = {
+  match: ReactRouterPropTypes.match.isRequired,
```
Comment (NikitaSedyx) [inline src/Receiving.js]: is it used?

File: src/common/resources/titleResources.js (hunk 1 above)
Comment (aliaksei-chumakou) [reply src/common/resources/titleResources.js]: right

File: src/TitleForm/TitleForm.js (hunk 2 above)
Comment (aliaksei-chumakou) [reply src/TitleForm/TitleForm.js]: oops, no

File: src/TitleForm/TitleForm.js (hunk 3 above)
Comment (aliaksei-chumakou) [reply src/TitleForm/TitleForm.js]: it's in ui-invoice

File: src/TitleForm/TitleFormContainer.js (hunk 4 above)
Comment (aliaksei-chumakou) [reply src/TitleForm/TitleFormContainer.js]: good question, null is simpler to use in any component, when LoadingPane return a Pane that requires a Paneset above in hierarchy

File: src/Receiving.js (hunk 5 above)
Comment (aliaksei-chumakou) [reply src/Receiving.js]: for now - no, but we will require this at the Edit route

File: src/common/resources/titleResources.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/common/resources/titleResources.js]: Don't forget about `CHANGELOG.md`

File: src/Receiving.js [hunk 6]
```diff
@@ -5,13 +5,24 @@ import {
...
+      />
+      <Route
+        component={ReceivingListContainer}
+        path="/receiving"
+      />
     </Switch>
   );
 };
 
+Receiving.propTypes = {
```
Comment (NikitaSedyx) [inline src/Receiving.js]: empty propTypes?

Comment (aliaksei-chumakou) [reply src/Receiving.js]: removed

**Pull request merged**